float64 xD
float64 yD
float64 thD
---
float64 x0
float64 y0
float64 th0
